app/models/user.rb

```diff
@@ -90,12 +90,12 @@ class User < ActiveRecord::Base
   paginates_per 15
 
   # Relationships
-  has_and_belongs_to_many :roles
-  has_many :topics
-  has_many :posts
-  has_many :votes
-  has_many :docs
-  has_many :api_keys
+
+  has_many :topics, dependent: :delete_all
+  has_many :posts, dependent: :delete_all
```

Comment:
or instead of a "dummy user": remove all the user related data and use a "generic name" for that user. otherwise a user can get a big bucket of related posts / topics which are not related.